services/ppac/src/main/java/app/coronawarn/datadonation/services/ppac/ios/verification/scenario/ratelimit/ProdPpacIosRateLimitStrategy.java

```diff
@@ -37,30 +38,19 @@ public void validateForEdus(ApiToken apiToken) {
   }
 
   /**
-   * Check Rate Limit for PPA Scenario. ApiToken in a PPA Scenario can only be used once a day.
+   * Check Rate Limit for PPA Scenario. ApiToken in a PPA Scenario can only be used once every 23 hours.
    *
    * @param apiToken the ApiToken that needs to be validated.
    */
   public void validateForPpa(ApiToken apiToken) {
     apiToken.getLastUsedPpac().ifPresent(getLastUsedEpochSecond -> {
-      LocalDate currentDateUtc = TimeUtils.getLocalDateForNow();
-      LocalDate lastUsedForPpaUtc = getLocalDateFor(getLastUsedEpochSecond);
-      logLastUpdate(getLastUsedEpochSecond, currentDateUtc, lastUsedForPpaUtc, Instant.now());
-
-      if (currentDateUtc.isEqual(lastUsedForPpaUtc)) {
+      LocalDateTime currentDateUtc = getLocalDateTimeForNow();
+      LocalDateTime lastUsedForPpaUtc = ofEpochSecond(getLastUsedEpochSecond).atOffset(UTC).toLocalDateTime();
+      long hours = HOURS.between(lastUsedForPpaUtc, currentDateUtc);
+      if (hours < VALIDITY_IN_HOURS) {
```

Comment:
Can I please have a brief summary of the logic behind the 23 hours validity instead of a day ? I find it strange.

Reply:
@alexmoroianu  Mobile Clients save the date last successful submission. Then for example 4 hours pass. They check if the difference between this new timestamp and the latest submission is greater or equal 23 hours if not they discard the request. So it can happen that the clients send one submission at 00:30 and then a second one at 23:33 for example. This would lead to an error in the backend. Thats why we decided to also use 23 hours

Reply:
Time varies with the mobiles. To have less 429 errors, we relax the constraint to "not reuse it within 23 hours" instead of "do not use it at the same day". See EXPOSUREBACK-1542 for more.